### DISCUSSION TITLE
Rebuild page/Invalidate cache using webhook

### DISCUSSION THREAD

**Discussion by PhilBookst** [flareact/flareact#192] [Q&A]
Hey,

is there a way to rebuild a statically generated page using a webhook from eg. a cms?

Say I want to make a change in a blog post and then a webhook is fired on save which either rebuilds the page or invalidates the cache and then builds the page afterwards?

Thanks for your help and props for the fantastic project!

**Comment by traffisco** [comment]
You probably can use KV storage to save page versions then invalidate the cache on version update. You can hook it to wrangler to update the version in KV on save. Sometimes along those lines.

**Reply by PhilBookst** [reply]
Do you have an example on how to invalidate the cache by any chance?

**Reply by traffisco** [reply]
Not really. You set a revalidate parameter in getEdgeProps, you can dynamically change it to force revalidation(ie using KV storage to compare page versions).
[caching-and-revalidation](https://flareact.com/docs/data-fetching#caching-and-revalidation)

**Reply by traffisco** [reply]
Actually, rethinking this one...
It wouldn't work. Once the revalidate parameter is set and the response is cached, it be pulled from cache for following requests and never hit the revalidate parameter until time is up.

**Reply by PhilBookst** [reply]
Exactly, that's what I was wondering about. So is there no way to delete/invalidate this cache? 

Maybe an api route or another worker which deletes the cache for a given page?
If the cache is deleted the next request would run getEdgeProps() again which would display the new content or am I missing something here?

**Reply by traffisco** [reply]
You can setup an api route that will do cache.delete. You’ll need to recreate the cache key(url+build number). 
https://github.com/flareact/flareact/blob/36cc6429dfb49bf08f8a5e098c1c9ebd893eb068/src/worker/index.js#L138

note that you have 2 urls, one “direct url” and one json props url to delete

**Reply by traffisco** [reply]
```
function getPropsCacheKey(request) {
  const url = new URL(request.url);
  const propsUrl = `${url.origin}/_flareact/props${url.pathname}.json${url.search}/${process.env.BUILD_ID}`;
  return new Request(new URL(propsUrl).toString(), request);
}

function getCacheKey(request) {
  const url = request.url + "/" + process.env.BUILD_ID;
  return new Request(new URL(url).toString(), request);
}
```

**Reply by PhilBookst** [reply]
Looks promising, will try it out soon! 

Appreciate your help!
